File: kotlinTutorials/kt10/src/main/java/com/java/kt10/LazyThreadSafeSynchronized.java
package com.java.kt10;

/**
 * 线程同步锁
 */
public class LazyThreadSafeSynchronized {
    private static LazyThreadSafeSynchronized instance;

    private LazyThreadSafeSynchronized() {

    }

    public static synchronized LazyThreadSafeSynchronized getInstance() {
        if (null == instance) {
            instance = new LazyThreadSafeSynchronized();
        }
        return instance;
    }
}

File: kotlinTutorials/kt10/src/main/kotlin/com/kotlin/kt10/LazyThreadSafeSynchronized.kt
package com.kotlin.kt10

/**
 * 线程同步锁
 */
class LazyThreadSafeSynchronized private constructor(){
    companion object {
        var instance: LazyThreadSafeSynchronized? = null

        @Synchronized
        fun getInstance2() : LazyThreadSafeSynchronized {
            if (null == instance) {
                instance = LazyThreadSafeSynchronized()
            }
            return instance!!
        }
    }
}